### PR TITLE
Implement 'POST /portfolioDrafts'

### PR DESCRIPTION
**Pull request by jasonburkert-ccpo**
Use as foundation to move forward on other tickets.

Implements `POST /portfolioDrafts`

Creates a new JSON document to be stored in Cosmos DB in Portfolios collection.  JSON will contain the following keys:

- `id` - a generated v4 UUID
- `status` - CSP provisioning status of _not_started_
- `created_at` - audit field with date/timestamp
- `updated_at` - audit field with date/timestamp

Ticket: AT-6399

### CODE REVIEW

File: provisioning/src/handlers/portfolioDrafts/portfolioDrafts.post.ts [hunk 1]
```diff
@@ -12,11 +12,11 @@ export async function postPortfolioDrafts (context: Context, req: HttpRequest):
     const { database } = await client.databases.createIfNotExists({ id: "atat" });
     const { container } = await database.containers.createIfNotExists({ id: "portfolios" });
 
-    // TODO: generate timestamps
+    let now = JSON.stringify(new Date());
```
Comment (jasonburkert-ccpo) [inline provisioning/src/handlers/portfolioDrafts/portfolioDrafts.post.ts:15]: Better as `let now = new Date().toISOString()`?  That's what underlying stringify.

File: provisioning/src/handlers/portfolioDrafts/post.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { Context, HttpRequest } from '@azure/functions'
+const { CosmosClient } = require("@azure/cosmos");
```
Comment (anhtran-ccpo) [inline provisioning/src/handlers/portfolioDrafts/post.ts:2]: Is there a reason for using require over import?

Comment (jasonburkert-ccpo) [reply provisioning/src/handlers/portfolioDrafts/post.ts:2]: I like the suggested ES6 syntax.  Makes it similar to the line above.

Comment (jeffsegal-ccpo) [reply provisioning/src/handlers/portfolioDrafts/post.ts:2]: Nope, probably just copypasta that we should fix.

File: provisioning/src/handlers/portfolioDrafts/post.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { v4 as uuidv4 } from 'uuid';
+import dayjs from "dayjs";
+import {PortfolioSummary} from "../../lib/models/PortfolioSummary";
+import {ProvisioningStatus} from "../../lib/models/ProvisioningStatus";
+
+export async function handler (context: Context, req: HttpRequest): Promise<void> {
```
Comment (anhtran-ccpo) [inline provisioning/src/handlers/portfolioDrafts/post.ts:8]: would this function break if we remove the req param? it is not being used

Comment (jeffsegal-ccpo) [reply provisioning/src/handlers/portfolioDrafts/post.ts:8]: Nope we should be able to remove it.

**Comment by anhtran-ccpo** [comment]
Will the shared infrastructure need to be copied over for all of apis we will have? I am wondering it it makes sense to keep it in provisioning if it is not unique.

File: provisioning/src/handlers/portfolioDrafts/post.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    const connectionString = process.env["COSMOS-CONNECTION-STRING"];
+    context.log("Connecting to " + connectionString)
+    const client = new CosmosClient(connectionString);
+
+    // TODO: create these in terraform & fail fast
```
Comment (anhtran-ccpo) [inline provisioning/src/handlers/portfolioDrafts/post.ts:13]: Are these TODOs still active?

Comment (jeffsegal-ccpo) [reply provisioning/src/handlers/portfolioDrafts/post.ts:13]: That particular one is half-active I guess. The DB and Container are in Terraform, but we do still want to fail fast if this code executes and the underlying resources aren't there.

**Comment by jeffsegal-ccpo** [comment]
To @anhtran-ccpo 's question on shared infrastructure - you're right that we probably ought to move it into a top-level folder in the monorepo since all Applications within it would likely use it (they may also have their own specific Terraform resources as well some day).

**Review by jasonburkert-ccpo — COMMENTED** [review]
Deployed shared infrastructure and new function app with @zachclark-ccpo in about 45 minutes.  Triggered new function with Postman, new record is in Cosmos DB portfolios collection.  Ready to merge.

image

**Review by anhtran-ccpo — APPROVED** [review]

**Review by zachclark-ccpo — APPROVED** [review]

**Pull request merged**